content/docs/1.1.1/advanced-resources/rancher-cluster-restore.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Restore cluster with a Rancher snapshot
+weight: 4
+---
+
+This doc describes what users need to do after restore the cluster with a Rancher snapshot.
+
+## Assumptions:
+- Most of the data still exist in the cluster before the restore and can be directly reused then.
+- There is a backupstore holding all volume data.
+- The setting [`Disable Revision Counter`](../../references/settings/#disable-revision-counter) is false. (It's false by default.) Otherwise, users need to manually check if the data among volume replicas are consistent, or directly restore volumes from backup.
+
+## Expectation:
+- All settings and node & disk configs will be restored.
+- As long as the valid data still exists, the volumes can be recovered without using backup. In other words, we will try to avoid restoring backups, which may help reduce Recovery Time Objective (RTO) as well as save bandwidth:
+- Detect the invalid or out-of-sync replicas as long as the related volume still contains a valid replica after the restore.
+
+## Behaviors & Requirement of Rancher restore
+- According to [the Rancher restore article](https://rancher.com/blog/2018/2018-05-30-recover-rancher-kubernetes-cluster-from-backup/), you have to restart the Kubernetes components on all nodes. Otherwise, there will be tons of resource update conflicts in Longhorn.
+
+## Actions after the restore
+- Restart all Kubernetes components for all nodes. See the above link for more details.
+
+- Kill all longhorn manager pods then Kubernetes will automatically restart them. Wait for conflicts in longhorn manager pods to disappear.
+
+- All volumes may be reattached. If a Longhorn volume is used by a single pod, users need to shut down then recreate it. For Deployments or Statefulsets, Longhorn will automatically kill then restart the related pods.
+    ```sh
+    docker restart kube-apiserver kubelet kube-controller-manager kube-scheduler kube-proxy
+    docker ps | grep flannel | cut -f 1 -d " " | xargs docker restart
+    docker ps | grep calico | cut -f 1 -d " " | xargs docker restart
+    ```
+
+- If the following happens after the snapshot and before the cluster restore:
+    - A volume is unchanged: Users don't need to do anything.
+    - The data is updated : Users don't need to do anything typically. Longhorn will automatically fail the replicas that don't contain the latest data.
+    - A new volume is created: This volume will disappear after the restore. Users need to recreate a new volume, launch [a single replica volume](../data-recovery/export-from-replica) based on the replica of the disappeared volume, then transfer the data to the new volume.
+    - A volume is deleted: Since the data is cleaned up when the volume is removed, the restored volume actually contains no data. Users may need to re-delete it.
```

Comment:
```suggestion
    - A volume is deleted: Since the data is cleaned up when the volume is removed, the restored volume contains no data. Users may need to re-delete it.
```